vspace/src/main/webapp/WEB-INF/views/exhibition/module.jsp

```diff
@@ -28,87 +28,147 @@ body {
+                <span class="tooltiptext">Go To Space</span>
+            </div>
+        </a>
+        <a
+            href="<c:url value="/exhibit/${spaceId}/module/${module.id}/sequence/${startSequenceId}" />">
+            <div class="exit_to_branchingPoint_Class">
+                <svg class="Ellipse_5">
+                        <ellipse fill="rgba(255,255,255,1)"
+                        class="Ellipse_5_Class" rx="22" ry="22" cx="22"
+                        cy="22">
+                        </ellipse>
+                    </svg>
+                <i class="fas fa-fast-backward fa-2x Icon_awesome_angle_double_left"></i>
+                <span class="tooltiptext">Go To Start Sequence of Module</span>
+            </div>
+        </a>
+        <c:if test="${showBackToPreviousChoice eq true}">
+        <a href="<c:url value="/exhibit/${spaceId}/module/${module.id}/sequence/${previousSequenceId}/slide/${previousBranchingPointId}?back=true" />">
+            <div class="exit_to_previousChoice_Class">
+                <svg class="Ellipse_5_be">
+              <ellipse fill="rgba(255,255,255,1)"
+                        class="Ellipse_5_be_Class" rx="22" ry="22"
+                        cx="22" cy="22">
+              </ellipse>
+            </svg>
+                <i class="fas fa-step-backward fa-2x Icon_awesome_angle_double_left"></i>
+                <span class="tooltiptext">Go To Previous Branching Choice</span>
             </div>
+        </a>
         </c:if>
-    </div>
-<!-- </div> -->
+        <div class="Group_8_Class">
+            <h3>${currentSlideCon.name}</h3>
+            <c:forEach items="${currentSlideCon.contents}"
+                var="contents">
+                <c:if
+                    test="${contents['class'].simpleName ==  'ImageBlock'}">
+                    <img id="${contents.id}" class="imgDiv"
+                        src="<c:url value="/api/image/${contents.image.id}" />" />
+                </c:if>
+                <c:if
+                    test="${contents['class'].simpleName ==  'TextBlock'}">
+                    <div id="${contents.id}" class="textDiv">
+                        <p>${contents.text}</p>
+                    </div>
+                </c:if>
+                <c:if
+                    test="${contents['class'].simpleName == 'ChoiceBlock'}">
+                    <div id="${contents.id}"
+                        class="textDiv"
+                        style="margin: 1%;">
+                        <c:if test="${contents.showsAll eq true}">
+                            <div class="list-group">
+                                <c:forEach items="${choices}"
+                                    var="choice">
+                                    <a
+                                        href="<c:url value="/exhibit/${spaceId}/module/${module.id}/sequence/${choice.sequence.id}/slide/${choice.sequence.slides[0].id}?previousSequenceId=${currentSequenceId}&branchingPoint=${slideId}" />"
```

Comment:
i would still link the sequence here as you had it before, just pass the parameters through to the slide redirect; if you do it this way, you will have to deal with empty sequences here instead of in the controller (where you should deal with it).